core/src/com/unciv/logic/automation/city/ConstructionAutomation.kt

```diff
@@ -57,8 +63,14 @@ class ConstructionAutomation(val cityConstructions: CityConstructions){
         choices.add(ConstructionChoice(choice, choiceModifier, cityConstructions.getRemainingWork(choice)))
     }
 
-    private fun Collection<INonPerpetualConstruction>.isBuildable(): Collection<INonPerpetualConstruction> {
-        return this.filter { it.isBuildable(cityConstructions) }
+    private fun Sequence<INonPerpetualConstruction>.isBuildable(): Sequence<INonPerpetualConstruction> {
```

Comment:
This should really be named filterBuildable. I know it was already like this, but if we're already fixing...

Reply:
That's a good idea.